.github/workflows/tests.yaml

```diff
@@ -6,10 +6,10 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-focal
```

Comment:
Use `ubuntu-20.04` here or stay with `ubuntu-latest`

Reply:
I don't think you need to pin the OS version of the runner. Unless `pip` compiles a Python module the OS version shouldn't really matter.

Reply:
For future reference, here is a list of supported OS names:

<https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories>

Reply:
Thanks for the review Nicolas. If I don't pin the version how will pip know that I want to run python 3.8/3.9?

Reply:
The `actions/setup-python` action will set up the correct Python version. When you then run something using `python3` or similar in the runner it will use the Python version you had set up with that action.

Reply:
Where would that go? I don't see that file or section in this project or in tests.yaml file.

Reply:
It's already there:

https://github.com/openstack-charmers/charm-layer-ovn/pull/89/files#diff-95557d53bf91069e59d82b9d8fcfaf52ec84a762858983edd5ef3c9b3e4c8191R17

Reply:
Ah ok, got it. So, if "python-version: [3.8, 3.9]" is set, "actions/setup-python@v2" will configure PIP for those two versions?

Reply:
Yes, it will create two VMs, one for each of the Python versions you put in that list.